api/clients/relay_client.go

```diff
@@ -45,49 +45,44 @@ type RelayClient interface {
 type relayClient struct {
 	config *RelayClientConfig
 
-	initOnce map[corev2.RelayKey]*sync.Once
-	conns    map[corev2.RelayKey]*grpc.ClientConn
-	logger   logging.Logger
-
-	grpcClients map[corev2.RelayKey]relaygrpc.RelayClient
+	// initOnce is used to ensure that the connection to each relay is initialized only once.
+	// It maps relay key to a sync.Once instance: `map[corev2.RelayKey]*sync.Once`
+	initOnce *sync.Map
+	// conns maps relay key to the gRPC connection: `map[corev2.RelayKey]*grpc.ClientConn`
+	conns  sync.Map
+	logger logging.Logger
+
+	// grpcClients maps relay key to the gRPC client: `map[corev2.RelayKey]relaygrpc.RelayClient`
+	grpcClients sync.Map
 }
 
 var _ RelayClient = (*relayClient)(nil)
 
 // NewRelayClient creates a new RelayClient that connects to the relays specified in the config.
 // It keeps a connection to each relay and reuses it for subsequent requests, and the connection is lazily instantiated.
```

Comment:
Why lazy init? It's not that expensive to create a connection once per Node creation

Reply:
Just preventing a case that creates a large number of connections. This client is not necessarily only used by the node as it's meant to be used as a library